data/nodes.ts

```diff
@@ -224,39 +221,59 @@ async function checkNodeStatus(endpoint: string, authentication?: string | null)
           params: [],
         }),
       })
-    }catch(err) {
-      return {
-        status: false,
-        loadTime: -1,
-      };
-    };
-    
     if (result.status !== 200) {
       console.log(endpoint, 'failed')
       console.log(await result.text());
       return {
-        status: false,
-        loadTime: -1,
+          status: false,
+          loadTime: -1,
       };
+    } else {
+      let loadTime = finish(start, endpoint);
+      console.log("success");
+      console.log(endpoint);
+      console.log(await result.text());
+      return {
+        status: true,
+        loadTime:loadTime
+      }
     }
-    const json = await result.json();
-
-    let loadTime = finish(start, endpoint);
-    return {
-      status: !!json.result,
-      loadTime
-    };
   } catch (e: any) {
     console.warn(e.message || e);
-
-    let loadTime = finish(start, endpoint);
     return {
       status: false,
-      loadTime
+      loadTime:-1,
     };
   }
 }
 
+const timer = (ms:number) => new Promise(res => setTimeout(res, ms))
+
+async function checkNodeStatus(endpoint: string, authentication?: string | null): Promise<Status> {
+  var totalLoadTime = 0
+  var rounds = 5
+  var timeInterval = 100 //ms
+  
+  for (var i = 0; i < rounds; i++) {
+      await timer(timeInterval)
+      var result = await getFetchTime(endpoint, authentication)
+      if (result.status){
+        totalLoadTime += result.loadTime
```

Comment:
Initial tests of this nature are often significantly higher in latency than following tests in quick succession (initial outputs are not representative of provider latency, and thus it is best practice to leave the first test in a multi-round testing method like this out of the final average). Example: Test 1, 1072ms; Test 2, 97ms; Test 3, 80ms; Test 4, 86ms; Test 5, 77ms; etc.

By keeping the first test, you'll run the chance of seeing the first result >1s, weighing down the following results. I'd suggest reworking this to run a first test, throw away that number, and then follow that up with the next 5 tests, then average those.

Reply:
This common issue in latency testing is also the reason for the original inaccuracy of some providers on the page.